### DISCUSSION TITLE
How can I connect Aurora serverless database to Next app

### DISCUSSION THREAD

**Discussion by adherb** [vercel/next.js#37119] [Help]
Hello,

I am trying to connect my Next app to an Aurora serverless database instance by I can't get it to work.

I am trying to use the RDS data api, but my Next api functions keep timing out.

Has anyone got any idea how I can achieve this?

Thanks,
Adam

**Comment by icyJoseph** [comment]
Hi,

Could you maybe show one of those api functions that is timing out?

To start, it would help to see at used packages, and code structure.

I also found this resource: https://www.jerrychang.ca/writing/aws-aurora-technical-series-postgresql-integration-nextjs, which perhaps could help you? but I am not sure it fully applies to your case.

**Reply by adherb** [reply]
Hi @icyJoseph, thanks for the reply.

Sure, here is a test function I've been trying:
```
export default async function handler(req, res) {
  try {
    let results = await data.query(`SELECT * FROM users LIMIT 4`);
    res.json({ results });
  } catch (err) {
    console.log("ERROR", err);
  }
}
```

I know it does work as shown below, but can't get it to work.

https://twitter.com/leeerob/status/1430177197105496071?lang=en

**Reply by icyJoseph** [reply]
And I guess the issue would be that only the try block returns, what error is it logging?

You do need to handle the error case, with a `res.status(400).json({error:'Bad request'})` for example, it depends on the type of failure.

Anyhow, what's the catch block logging?
